aspnetcore/fundamentals/hosted-services.md

```diff
@@ -0,0 +1,87 @@
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 02/12/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: fundamentals/hosted-services
+---
+# Background tasks with hosted services in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+In ASP.NET Core, background tasks are implemented as *hosted services*. A hosted service is a class with background task logic that implements the [IHostedService](/dotnet/api/microsoft.extensions.hosting.ihostedservice) interface. This topic provides three hosted service examples:
+
+* Background task that runs on a timer.
+* Hosted service that activates a scoped service. The scoped service can use dependency injection.
+* Queued background tasks that run sequentially.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/hosted-services/samples/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## IHostedService interface
+
+Hosted services implement the [IHostedService](/dotnet/api/microsoft.extensions.hosting.ihostedservice) interface. The interface defines two methods for objects that are managed by the host:
+
+* [StartAsync(CancellationToken)](/dotnet/api/microsoft.extensions.hosting.ihostedservice.startasync) - Called after the server has started and [IApplicationLifetime.ApplicationStarted](/dotnet/api/microsoft.aspnetcore.hosting.iapplicationlifetime.applicationstarted) is triggered. `StartAsync` contains the logic to start the background task.
+
+* [StopAsync(CancellationToken)](/dotnet/api/microsoft.extensions.hosting.ihostedservice.stopasync) - Triggered when the host is performing a graceful shutdown. `StopAsync` contains the logic to end the background task and dispose of any unmanaged resources. If the app shuts down unexpectedly (for example, the app's process fails), `StopAsync` might not be called.
+
+The hosted service is a singleton that's activated once at app startup and gracefully shutdown at app shutdown. When [IDisposable](/dotnet/api/system.idisposable) is implemented, resources can be disposed when the service container is disposed. If an error is thrown during background task execution, `Dispose` should be called even if `StopAsync` isn't called.
+
+## Timed background tasks
+
+A timed background task makes use of the [System.Threading.Timer](/dotnet/api/system.threading.timer) class. The timer triggers the task's `DoWork` method. The timer is disabled on `StopAsync` and disposed when the service container is disposed on `Dispose`:
+
+[!code-csharp[](hosted-services/sample/Services/TimedHostedService.cs?name=snippet1&highlight=9,23,30)]
+
+The service is registered in `Startup.ConfigureServices`:
+
+[!code-csharp[](hosted-services/sample/Startup.cs?name=snippet1)]
+
+## Consuming a scoped service in a background task
+
+To use scoped services within an `IHostedService`, create a scope. No scope is needed for a hosted service by default.
```

Comment:
No scope is created, I wouldn't say needed. That's implied by the fact that it's a single.